docs/DeveloperManuals/Project.md

```diff
@@ -0,0 +1,320 @@
+| project_1        | Board     | jira:JiraBoard:1:lake    |
+| project_2        | Repo      | github:GithubRepo:1:lake |
+
+# How to operate the table through the API
+
+By visiting `[Your Config-UI Host]/api/swagger/index.html`, you can go to the swagger document of `Devlake` deployed. In this document, `framework/projects` `framework/ProjectMetrics`, and `framework/plugins ` is the related API of `project`.
+
+To protect the integrity of the data in the database as much as possible, we recommend using the API to perform corresponding operations instead of directly using SQL statements to operate the tables in the database.
+
+When we need to manipulate the `projects` table, we can use the `framework/projects`:
+- We can create a new `project` object through the `POST` `/projects` interface, which requires a `projectName` field as its name, which is unique.
+
+After we created the `project` object or we already had a `project`:
+- We can use the `GET /projects/:projectName` interface to obtain specific `project` information. In addition to the name, this information also includes the update time and creation time of the project, and will be associated with all The relevant `project_metrics` and `project_mapping` information of the current `project` will be returned after integration.
+- We can use the `PATCH /projects/:projectName` interface to update the information of a specific `project`, such as updating its related `description` information.
+- We can get all `project` information through the `GET /projects` interface. The returned information can be filtered by setting the parameter `search`, and the returned information can be paged by setting the parameters `page`, and `pageSize`.
+
+After we created the `project` object or we already had a `project`:
+- we can use the `POST /project_metrics/:projectName/:pluginName` to create a set of relationships between `project` and `Metrics Plugin`, Use the `projectName` and `pluginName` we obtained earlier to build this relationship. At the same time configure the corresponding `option` for it.
+
+After we create a `project_metrics` relation or we already had one:
+- We can use the `GET /project_metrics/:projectName/:pluginName` to get a set of `project_metrics` information.
+- We can use the `PATCH /project_metrics/:projectName/:pluginName` to update the `project_metrics` information.
+
+If we need to know which plugins are `Metrics Plugin`:
+- we can use the `GET` `/plugins`, it will return a list of all plugin names and information about the plugins and `Metrics Plugin`.
+
+# The interface that needs to be implemented
+
+We divide plugins into three categories
+- The first category is `Base Plugin`, such as `core`, `helper`, etc. These plug-ins provide basic support services for other plug-ins.
+- The second category is `Data Source Plugin`, such as `GitLab` `GitHub` `Jira` `Jenkins`, etc. These plugins collect data from various data sources
```

Comment:
Combine this information(Line133-134) to the "Interface" section

Reply:
this is arealdy on "Interface" section